tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs

```diff
@@ -203,11 +203,45 @@ private static string TransformDocTemplate(ApiMetadata api, string text)
 The credentials will automatically be used to authenticate. See the [Getting Started With
 Authentication](https://cloud.google.com/docs/authentication/getting-started) guide for more details.";
 
+            string clientClasses = CreateClientClassesDocumentation(api);
+
+            string clientConstruction =
+@"Create a client instance by calling the static `Create` method, optionally
+specifying an end-point or channel and settings";
+
             return text
                 .Replace("{{title}}", title)
                 .Replace("{{description}}", description)
                 .Replace("{{installation}}", installation)
-                .Replace("{{auth}}", auth);
+                .Replace("{{auth}}", auth)
+                .Replace("{{client-classes}}", clientClasses)
+                .Replace("{{client-construction}}", clientConstruction);
+        }
+
+        private static string CreateClientClassesDocumentation(ApiMetadata api)
+        {
+            if (api.Type != "grpc")
+            {
+                return "FIXME"; // No automatic templating for this API
+            }
+            var layout = DirectoryLayout.ForApi(api.Id);
+            var packageSource = Path.Combine(layout.SourceDirectory, api.Id);
+            var sourceFiles = Directory.GetFiles(packageSource, "*Client.cs");
+            var clients = sourceFiles
+                .Where(file => File.ReadAllText(file).Contains(": ServiceSettingsBase")) // Check it contains a generated client
```

Comment:
A tad crude/unreliable perhaps? Maybe a TODO to improve robustness of generated client detection?

Reply:
Will add a TODO, but I strongly suspect it'll be accurate enough forever :)